cmd/admin-heal.go

```diff
@@ -72,304 +76,556 @@ EXAMPLES:
+	case madmin.HealItemMultipartUpload:
+		return fmt.Sprintf("[multipart-upload]%s/%s(%s)", h.Bucket,
+			h.Object, h.Detail)
+	}
+	return "** unexpected **"
 }
 
-// String colorized service status message.
-func (u healListMessage) String() string {
-	msg := ""
-	var healStatus madmin.HealStatus
-
-	// Check if we have object heal information
-	if u.Object != nil {
-		msg += fmt.Sprintf("Object: %s/%s, ", u.Bucket, u.Object.Key)
-		healStatus = u.Object.HealObjectInfo.Status
-	} else {
-		msg += fmt.Sprintf("Upload: %s/%s, ", u.Bucket, u.Upload.Key)
-		healStatus = u.Upload.HealUploadInfo.Status
-	}
-
-	// Print heal status
-	switch healStatus {
-	case madmin.CanHeal:
-		msg += "can be healed."
-	case madmin.CanPartiallyHeal:
-		msg += "can be partially healed."
-	case madmin.Corrupted:
-		msg += "cannot be healed."
-	case madmin.QuorumUnavailable:
-		msg += "quorum not available for healing."
-	}
-	return console.Colorize("Heal", msg)
+func (h hri) getHRTypeAndName() (typ, name string) {
+	name = fmt.Sprintf("%s/%s", h.Bucket, h.Object)
+	switch h.Type {
+	case madmin.HealItemMetadata:
+		typ = "system"
+		name = h.Detail
+	case madmin.HealItemBucketMetadata:
+		typ = "system"
+		name = "bucket-metadata:" + name
+	case madmin.HealItemMultipartUpload:
+		typ = "object"
+		name = "ongoing-upload:" + name
+	case madmin.HealItemBucket:
+		typ = "bucket"
+	case madmin.HealItemObject:
+		typ = "object"
+	default:
+		typ = fmt.Sprintf("!! Unknown heal result record %#v !!", h)
+		name = typ
+	}
+	return
 }
 
-// JSON jsonified service status Message message.
-func (u healListMessage) JSON() string {
-	u.Status = "success"
-	statusJSONBytes, e := json.Marshal(u)
-	fatalIf(probe.NewError(e), "Unable to marshal into JSON.")
+func (h hri) getHealResultStr() string {
+	typ, name := h.getHRTypeAndName()
 
-	return string(statusJSONBytes)
+	switch h.Type {
+	case madmin.HealItemMetadata, madmin.HealItemBucketMetadata:
+		return typ + ":" + name
+	default:
+		return name
+	}
 }
 
-// healBucketListMessage container to hold heal information.
-type healBucketListMessage struct {
-	Status string            `json:"status"`
-	Bucket madmin.BucketInfo `json:"bucket"`
+// lineTrunc - truncates a string by adding ellipsis in the middle
+func lineTrunc(content string, maxLen int) string {
+	runes := []rune(content)
+	rlen := len(runes)
+	if rlen <= maxLen {
+		return content
+	}
+	halfLen := maxLen / 2
+	fstPart := string(runes[0:halfLen])
+	sndPart := string(runes[rlen-halfLen:])
+	return fstPart + "…" + sndPart
 }
 
-// String colorized service status message.
-func (u healBucketListMessage) String() string {
-	msg := fmt.Sprintf("Bucket: `%s`, ", u.Bucket.Name)
-	switch u.Bucket.HealBucketInfo.Status {
-	case madmin.CanHeal:
-		msg += "can be healed"
-	case madmin.Corrupted:
-		msg += "cannot be healed"
-	case madmin.QuorumUnavailable:
-		msg += "quorum not available for healing"
-	}
-	msg += ".\n"
-	return console.Colorize("Heal", msg)
+type uiData struct {
```

Comment:
Can you move uiData code to a new source file ?

Reply:
Done.